### ISSUE TITLE
problems

### ISSUE THREAD

**Issue by Shakir-Alshehabi** [thomasbi1/cyberrunner#3]
when i try to train it gave me this problem blew this is new when I was trying to train before it was working this problem happened after the last update to the env.py   

[libprotobuf FATAL external/com_google_protobuf/src/google/protobuf/stubs/common.cc:83] This program was compiled against version 3.12.4 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.21.9).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "./obj-x86_64-linux-gnu/modules/dnn/versions.pb.cc".)
terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  This program was compiled against version 3.12.4 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.21.9).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "./obj-x86_64-linux-gnu/modules/dnn/versions.pb.cc".)

**Comment by thomasbi1** [comment]
That's strange. Is that the full error trace? Seems like an issue with the protobuf version, and not due to any code changes in env.py

**Comment by Shakir-Alshehabi** [comment]
this is the full execution of train something to note the problem may be connected to "jax[cuda12_pip] "  update. 
"  ros2 run cyberrunner_dreamer train

Config:
seed:                       1                                           (int)
method:                     name                                        (str)
task:                       gym_cyberrunner_dreamer:cyberrunner-ros-v0  (str)
logdir:                     ~/cyberrunner_logs/robust_2                 (str)
replay:                     cyberrunner                                 (str)
replay_size:                1000000.0                                   (float)
replay_online:              False                                       (bool)
eval_dir:                                                               (str)
filter:                     .*                                          (str)
jax.platform:               gpu                                         (str)
jax.jit:                    True                                        (bool)
jax.precision:              float16                                     (str)
jax.prealloc:               True                                        (bool)
jax.debug_nans:             False                                       (bool)
jax.logical_cpus:           8                                           (int)
jax.debug:                  False                                       (bool)
jax.policy_devices:         [0]                                         (ints)
jax.train_devices:          [0]                                         (ints)
jax.metrics_every:          10                                          (int)
run.script:                 train                                       (str)
run.steps:                  10000000000.0                               (float)
run.expl_until:             0                                           (int)
run.log_every:              10                                          (int)
run.save_every:             20                                          (int)
run.eval_every:             1000000.0                                   (float)
run.eval_initial:           True                                        (bool)
run.eval_eps:               1                                           (int)
run.eval_samples:           1                                           (int)
run.train_ratio:            128.0                                       (float)
run.train_fill:             0                                           (int)
run.eval_fill:              0                                           (int)
run.log_zeros:              False                                       (bool)
run.log_keys_video:         [image]                                     (strs)
run.log_keys_sum:           ^log_reward$                                (str)
run.log_keys_mean:          (log_entropy)                               (str)
run.log_keys_max:           ^$                                          (str)
run.from_checkpoint:                                                    (str)
run.sync_every:             1                                           (int)
run.actor_addr:             ipc:///tmp/5552                             (str)
run.actor_batch:            1                                           (int)
envs.amount:                1                                           (int)
envs.parallel:              process                                     (str)
envs.length:                0                                           (int)
envs.reset:                 True                                        (bool)
envs.restart:               True                                        (bool)
envs.discretize:            0                                           (int)
envs.checks:                False                                       (bool)
wrapper.length:             10                                          (int)
wrapper.reset:              True                                        (bool)
wrapper.discretize:         0                                           (int)
wrapper.checks:             False                                       (bool)
env.atari.size:             [64, 64]                                    (ints)
env.atari.repeat:           4                                           (int)
env.atari.sticky:           True                                        (bool)
env.atari.gray:             False                                       (bool)
env.atari.actions:          all                                         (str)
env.atari.lives:            unused                                      (str)
env.atari.noops:            0                                           (int)
env.atari.resize:           opencv                                      (str)
env.dmlab.size:             [64, 64]                                    (ints)
env.dmlab.repeat:           4                                           (int)
env.dmlab.episodic:         True                                        (bool)
env.minecraft.size:         [64, 64]                                    (ints)
env.minecraft.break_speed:  100.0                                       (float)
env.dmc.size:               [64, 64]                                    (ints)
env.dmc.repeat:             2                                           (int)
env.dmc.camera:             -1                                          (int)
env.loconav.size:           [64, 64]                                    (ints)
env.loconav.repeat:         2                                           (int)
env.loconav.camera:         -1                                          (int)
task_behavior:              Greedy                                      (str)
expl_behavior:              None                                        (str)
batch_size:                 4                                           (int)
batch_length:               64                                          (int)
data_loaders:               8                                           (int)
grad_heads:                 [decoder, reward, cont]                     (strs)
rssm.deter:                 2048                                        (int)
rssm.units:                 768                                         (int)
rssm.stoch:                 32                                          (int)
rssm.classes:               32                                          (int)
rssm.act:                   silu                                        (str)
rssm.norm:                  layer                                       (str)
rssm.initial:               learned                                     (str)
rssm.unimix:                0.01                                        (float)
rssm.unroll:                False                                       (bool)
rssm.action_clip:           1.0                                         (float)
rssm.winit:                 normal                                      (str)
rssm.fan:                   avg                                         (str)
encoder.mlp_keys:           states|goal$                                (str)
encoder.cnn_keys:           image                                       (str)
encoder.act:                silu                                        (str)
encoder.norm:               layer                                       (str)
encoder.mlp_layers:         4                                           (int)
encoder.mlp_units:          768                                         (int)
encoder.cnn:                resnet                                      (str)
encoder.cnn_depth:          64                                          (int)
encoder.cnn_blocks:         0                                           (int)
encoder.resize:             stride                                      (str)
encoder.winit:              normal                                      (str)
encoder.fan:                avg                                         (str)
encoder.symlog_inputs:      True                                        (bool)
encoder.minres:             4                                           (int)
decoder.mlp_keys:           states|goal$                                (str)
decoder.cnn_keys:           image                                       (str)
decoder.act:                silu                                        (str)
decoder.norm:               layer                                       (str)
decoder.mlp_layers:         4                                           (int)
decoder.mlp_units:          768                                         (int)
decoder.cnn:                resnet                                      (str)
decoder.cnn_depth:          64                                          (int)
decoder.cnn_blocks:         0                                           (int)
decoder.image_dist:         mse                                         (str)
decoder.vector_dist:        symlog_mse                                  (str)
decoder.inputs:             [deter, stoch]                              (strs)
decoder.resize:             stride                                      (str)
decoder.winit:              normal                                      (str)
decoder.fan:                avg                                         (str)
decoder.outscale:           1.0                                         (float)
decoder.minres:             4                                           (int)
decoder.cnn_sigmoid:        False                                       (bool)
reward_head.layers:         4                                           (int)
reward_head.units:          768                                         (int)
reward_head.act:            silu                                        (str)
reward_head.norm:           layer                                       (str)
reward_head.dist:           symlog_disc                                 (str)
reward_head.outscale:       0.0                                         (float)
reward_head.outnorm:        False                                       (bool)
reward_head.inputs:         [deter, stoch]                              (strs)
reward_head.winit:          normal                                      (str)
reward_head.fan:            avg                                         (str)
reward_head.bins:           255                                         (int)
cont_head.layers:           4                                           (int)
cont_head.units:            768                                         (int)
cont_head.act:              silu                                        (str)
cont_head.norm:             layer                                       (str)
cont_head.dist:             binary                                      (str)
cont_head.outscale:         1.0                                         (float)
cont_head.outnorm:          False                                       (bool)
cont_head.inputs:           [deter, stoch]                              (strs)
cont_head.winit:            normal                                      (str)
cont_head.fan:              avg                                         (str)
loss_scales.image:          1.0                                         (float)
loss_scales.vector:         1.0                                         (float)
loss_scales.reward:         1.0                                         (float)
loss_scales.cont:           1.0                                         (float)
loss_scales.dyn:            0.5                                         (float)
loss_scales.rep:            0.1                                         (float)
loss_scales.actor:          1.0                                         (float)
loss_scales.critic:         1.0                                         (float)
loss_scales.slowreg:        1.0                                         (float)
dyn_loss.impl:              kl                                          (str)
dyn_loss.free:              1.0                                         (float)
rep_loss.impl:              kl                                          (str)
rep_loss.free:              1.0                                         (float)
model_opt.opt:              adam                                        (str)
model_opt.lr:               0.0001                                      (float)
model_opt.eps:              1e-08                                       (float)
model_opt.clip:             1000.0                                      (float)
model_opt.wd:               0.0                                         (float)
model_opt.warmup:           0                                           (int)
model_opt.lateclip:         0.0                                         (float)
actor.layers:               4                                           (int)
actor.units:                768                                         (int)
actor.act:                  silu                                        (str)
actor.norm:                 layer                                       (str)
actor.minstd:               0.1                                         (float)
actor.maxstd:               1.0                                         (float)
actor.outscale:             1.0                                         (float)
actor.outnorm:              False                                       (bool)
actor.unimix:               0.01                                        (float)
actor.inputs:               [deter, stoch]                              (strs)
actor.winit:                normal                                      (str)
actor.fan:                  avg                                         (str)
actor.symlog_inputs:        False                                       (bool)
critic.layers:              4                                           (int)
critic.units:               768                                         (int)
critic.act:                 silu                                        (str)
critic.norm:                layer                                       (str)
critic.dist:                symlog_disc                                 (str)
critic.outscale:            0.0                                         (float)
critic.outnorm:             False                                       (bool)
critic.inputs:              [deter, stoch]                              (strs)
critic.winit:               normal                                      (str)
critic.fan:                 avg                                         (str)
critic.bins:                255                                         (int)
critic.symlog_inputs:       False                                       (bool)
actor_opt.opt:              adam                                        (str)
actor_opt.lr:               3e-05                                       (float)
actor_opt.eps:              1e-05                                       (float)
actor_opt.clip:             100.0                                       (float)
actor_opt.wd:               0.0                                         (float)
actor_opt.warmup:           0                                           (int)
actor_opt.lateclip:         0.0                                         (float)
critic_opt.opt:             adam                                        (str)
critic_opt.lr:              3e-05                                       (float)
critic_opt.eps:             1e-05                                       (float)
critic_opt.clip:            100.0                                       (float)
critic_opt.wd:              0.0                                         (float)
critic_opt.warmup:          0                                           (int)
critic_opt.lateclip:        0.0                                         (float)
actor_dist_disc:            onehot                                      (str)
actor_dist_cont:            normal                                      (str)
actor_grad_disc:            reinforce                                   (str)
actor_grad_cont:            backprop                                    (str)
critic_type:                vfunction                                   (str)
imag_horizon:               15                                          (int)
imag_unroll:                False                                       (bool)
horizon:                    333                                         (int)
return_lambda:              0.95                                        (float)
critic_slowreg:             logprob                                     (str)
slow_critic_update:         1                                           (int)
slow_critic_fraction:       0.02                                        (float)
retnorm.impl:               perc_ema                                    (str)
retnorm.decay:              0.99                                        (float)
retnorm.max:                1.0                                         (float)
retnorm.perclo:             5.0                                         (float)
retnorm.perchi:             95.0                                        (float)
actent:                     0.0003                                      (float)
expl_rewards.extr:          1.0                                         (float)
expl_rewards.disag:         0.1                                         (float)
expl_opt.opt:               adam                                        (str)
expl_opt.lr:                0.0001                                      (float)
expl_opt.eps:               1e-05                                       (float)
expl_opt.clip:              100.0                                       (float)
expl_opt.wd:                0.0                                         (float)
expl_opt.warmup:            0                                           (int)
disag_head.layers:          4                                           (int)
disag_head.units:           768                                         (int)
disag_head.act:             silu                                        (str)
disag_head.norm:            layer                                       (str)
disag_head.dist:            mse                                         (str)
disag_head.outscale:        1.0                                         (float)
disag_head.inputs:          [deter, stoch, action]                      (strs)
disag_head.winit:           normal                                      (str)
disag_head.fan:             avg                                         (str)
disag_target:               [stoch]                                     (strs)
disag_models:               8                                           (int)
CYBERRUNNER Replay buffer
[libprotobuf FATAL external/com_google_protobuf/src/google/protobuf/stubs/common.cc:83] This program was compiled against version 3.12.4 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.21.9).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "./obj-x86_64-linux-gnu/modules/dnn/versions.pb.cc".)
terminate called after throwing an instance of 'google::protobuf::FatalException'
  what():  This program was compiled against version 3.12.4 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.21.9).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "./obj-x86_64-linux-gnu/modules/dnn/versions.pb.cc".)"

**Comment by Shakir-Alshehabi** [comment]
I tried everything  to change protobuf and it still gave me this problem from sudo reinstall to pip protobuf installation to protobuf system removal and package installing

**Comment by martin20032001** [comment]
Were you able to identify the cause and resolve the issue?

**Comment by Shakir-Alshehabi** [comment]
I install protobuf as pip and apt and add it to CMakeLists.txt in cyberrunner_interface by  
//
set(Protobuf_INCLUDE_DIR "/usr/include")      
set(Protobuf_LIBRARIES "/usr/lib/x86_64-linux-gnu/libprotobuf.so")
set(Protobuf_PROTOC_EXECUTABLE "/usr/bin/protoc")
set(Protobuf_PROTOC_INCLUDE_DIR "/usr/include")
set(Protobuf_PROTOC_LIBRARY_DIR "/usr/lib/x86_64-linux-gnu")
set(Protobuf_PROTOC_LIBRARY "/usr/lib/x86_64-linux-gnu/libprotobuf.so")
set(Protobuf_PROTOC_EXECUTABLE "/usr/bin/protoc")
//
that what solved it for me.
one note to add you need to update requirements.txt in dremerv3 to the new one from the original dremerv3, becuse the new jax and jaxlib update doesn't include cuda12_pip.
